File: app/src/main/java/org/worshipsongs/service/UserPreferenceSettingService.java
package org.worshipsongs.service;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import org.worshipsongs.CommonConstants;
import org.worshipsongs.WorshipSongApplication;

import java.util.Map;

/**
 * Author: Seenivasan
 * Version: 1.0.0
 */
public class UserPreferenceSettingService
{

    private Context context = WorshipSongApplication.getContext();
    private SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);

    public UserPreferenceSettingService()
    {

    }

    public float getPortraitFontSize()
    {
        return sharedPreferences.getInt(CommonConstants.PRIMARY_FONT_SIZE_KEY, 18);
    }

    public float getLandScapeFontSize()
    {
        return sharedPreferences.getInt(CommonConstants.PRESENTATION_FONT_SIZE_KEY, 25);
    }

    public int getPrimaryColor()
    {
        Map<String, ?> all = sharedPreferences.getAll();
        int color;
        if (all.containsKey("primaryColor")) {
            color = Integer.parseInt(all.get("primaryColor").toString());
        } else {
            color = -12303292;
        }
        return color;
    }

    public int getSecondaryColor()
    {
        Map<String, ?> all = sharedPreferences.getAll();
        int color;
        if (all.containsKey("secondaryColor")) {
            color = Integer.parseInt(all.get("secondaryColor").toString());
        } else {
            color = -65536;
        }
        return color;
    }

    public int getPresentationBackgroundColor()
    {
        return sharedPreferences.getInt("presentationBackgroundColor", 0xff000000);
    }

    public int getPresentationPrimaryColor()
    {
        return sharedPreferences.getInt("presentationPrimaryColor", 0xffffffff);
    }

    public int getPresentationSecondaryColor()
    {
        return sharedPreferences.getInt("presentationSecondaryColor", 0xffffff00);
    }

    public boolean isKeepAwake()
    {
        return sharedPreferences.getBoolean("prefKeepAwakeOn", false);
    }

    public boolean isPlayVideo()
    {
        return sharedPreferences.getBoolean("prefVideoPlay", true);
    }

    public boolean isTamilLyrics()
    {
        return sharedPreferences.getBoolean("displayTamilLyrics", true);
    }

    public boolean isRomanisedLyrics()
    {
        return sharedPreferences.getBoolean("displayRomanisedLyrics", true);
    }

    public boolean isTamil()
    {
        return sharedPreferences.getInt(CommonConstants.LANGUAGE_INDEX_KEY, 0) == 0;
    }
}

File: app/src/main/java/org/worshipsongs/service/UserPreferenceSettingService.kt
package org.worshipsongs.service

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

import org.worshipsongs.CommonConstants
import org.worshipsongs.WorshipSongApplication

/**
 * Author: Seenivasan
 * Version: 1.0.0
 */
class UserPreferenceSettingService
{

    private val context = WorshipSongApplication.context
    private var sharedPreferences: SharedPreferences? = null

    val portraitFontSize: Float
        get() = sharedPreferences!!.getInt(CommonConstants.PRIMARY_FONT_SIZE_KEY, 18).toFloat()

    val landScapeFontSize: Float
        get() = sharedPreferences!!.getInt(CommonConstants.PRESENTATION_FONT_SIZE_KEY, 25).toFloat()

    val primaryColor: Int
        get()
        {
            val all = sharedPreferences!!.all
            val color: Int
            if (all.containsKey("primaryColor"))
            {
                color = Integer.parseInt(all["primaryColor"]!!.toString())
            } else
            {
                color = -12303292
            }
            return color
        }

    val secondaryColor: Int
        get()
        {
            val all = sharedPreferences!!.all
            val color: Int
            if (all.containsKey("secondaryColor"))
            {
                color = Integer.parseInt(all["secondaryColor"]!!.toString())
            } else
            {
                color = -65536
            }
            return color
        }

    val presentationBackgroundColor: Int
        get() = sharedPreferences!!.getInt("presentationBackgroundColor", -0x1000000)

    val presentationPrimaryColor: Int
        get() = sharedPreferences!!.getInt("presentationPrimaryColor", -0x1)

    val presentationSecondaryColor: Int
        get() = sharedPreferences!!.getInt("presentationSecondaryColor", -0x100)

    val isKeepAwake: Boolean
        get() = sharedPreferences!!.getBoolean("prefKeepAwakeOn", false)

    val isPlayVideo: Boolean
        get() = sharedPreferences!!.getBoolean("prefVideoPlay", true)

    val isTamilLyrics: Boolean
        get() = sharedPreferences!!.getBoolean("displayTamilLyrics", true)

    val isRomanisedLyrics: Boolean
        get() = sharedPreferences!!.getBoolean("displayRomanisedLyrics", true)

    val isTamil: Boolean
        get() = sharedPreferences!!.getInt(CommonConstants.LANGUAGE_INDEX_KEY, 0) == 0

    val displaySongBook: Boolean
        get() = sharedPreferences!!.getBoolean("prefDisplaySongbook", false)

    constructor()
    {
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
    }

    constructor(context: Context)
    {
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
    }
}
